src/mpid/ch4/src/ch4_recv.h

```diff
@@ -129,26 +129,13 @@ MPL_STATIC_INLINE_PREFIX int MPID_Recv_init(void *buf,
         MPIDI_NM_mpi_recv_init(buf, count, datatype, rank, tag, comm, context_offset, av, request);
 #else
     if (unlikely(rank == MPI_ANY_SOURCE)) {
-        mpi_errno =
-            MPIDI_SHM_mpi_recv_init(buf, count, datatype, rank, tag, comm, context_offset, request);
-
-        if (mpi_errno != MPI_SUCCESS) {
-            MPIR_ERR_POP(mpi_errno);
-        }
-
-        mpi_errno =
-            MPIDI_NM_mpi_recv_init(buf, count, datatype, rank, tag, comm, context_offset, av,
-                                   &(MPIDI_CH4I_REQUEST_ANYSOURCE_PARTNER(*request)));
-
-        if (mpi_errno != MPI_SUCCESS) {
-            MPIR_ERR_POP(mpi_errno);
+        /* NM_Irecv for ANY_SOURCE will also post a receive for SHM. */
+        mpi_errno = MPIDI_NM_mpi_recv_init(buf, count, datatype, rank, tag,
```

Comment:
What if SHM and NM need initialize differently?

I would rather see a fix that prevents `MPI_Start` double post than a solution that restricts on functionality.